src/rastervision/utils/files.py

```diff
@@ -146,8 +144,17 @@ def _sync_dir(delete=True):
         # th contents of output_dir since there's nothing there yet.
         _sync_dir(delete=False)
 
-
 # Ensure that RV temp directory exists. We need to use a custom location for
 # the temporary directory so it will be mirrored on the host file system which
 # is needed for running in a Docker container with limited space on EC2.
-make_dir(RV_TEMP_DIR)
+RV_TEMP_DIR = '/opt/data/tmp/'
+# no explicitly set tempdir
+if (not any([k in os.environ for k in ['TMPDIR', 'TEMP', 'TMP']]) and
+    not tempfile.tempdir):
+    tempfile.tempdir = RV_TEMP_DIR
+    make_dir(RV_TEMP_DIR)
+
+# now, ensure uniqueness for this process
+# the host may be running more than one rastervision process
+RV_TEMP_DIR = tempfile.mkdtemp()
```

Comment:
My impression is that this is supposed to set `RV_TEMP_DIR` to the value of the env var `TMP` when it has been set. This does not appear to happen though in Python 3.5.1. 

```
root@7a96485183fd:/opt/src# TMP='/opt/data/testing123' python
Python 3.5.1 |Continuum Analytics, Inc.| (default, Jun 15 2016, 15:32:45)
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import tempfile
>>> tempfile.mkdtemp()
'/tmp/tmpqxmmyzyz'
>>> import os
>>> os.environ['TMP']
'/opt/data/testing123'
```

Reply:
Yes, I am using this code (and generally at the env level) throughout the data science team. As for your test case that I'm also experimenting with, yes... I see the same behavior on 3.6...

Ah... I see what's going on... evidently, Python tries to be smart about it:

https://hg.python.org/cpython/file/tip/Lib/tempfile.py#l188

The user of the process must be able to write files to the directory! Which also means the directory must exist!

Reply:
Ok. Do you want to automatically create the directory then to be more consistent with how the rest of RV works?

Reply:
Sorry. Can you elaborate on your question? Are you asking about getting rid of the hard-coded `/opt/data/tmp/`? I'm trying to understand "automatically create the directory" phrase and the code does automatically create a temporary directory for RV's use...

Reply:
Since the directory pointed to by `TMP` must exist for `tempfile` to utilize it (as you discovered), that just raises the question of whether we should create that directory if it doesn't already exist. This would prevent any confusion (like I had, not realizing that the directory needs to exist) although it seems kind of wrong to override Python's behavior in this case.

Reply:
That's a good question. I would say "yes" but if `shutil.mktree()` fails, leaving a message (`warnings.warn`?) and let `tempfile` do what it would default to

Reply:
I think 4cec286 does what we want...